File: java/decorator/StarbuzzCoffee/decorator/condiments/Whip.java
package decorator.condiments;

import component.Beverage;
import decorator.CondimentDecorator;

public class Whip extends CondimentDecorator {

    private Beverage beverage;

    public Whip(Beverage beverage) { this.beverage = beverage; }

    @Override
    public double cost() { return beverage.cost() + 0.10; }

    @Override
    public String getDescription() { return beverage.getDescription() + ", Whip"; }
}

File: kotlin/decorator/StarbuzzCoffee/decorator/condiments/Whip.kt
package decorator.condiments

import component.Beverage
import decorator.CondimentDecorator

class Whip(beverage: Beverage) : CondimentDecorator() {

    private var beverage: Beverage? = null

    init { this.beverage = beverage }

    override fun getDescription(): String { return beverage!!.getDescription() + ", Whip" }

    override fun cost(): Double { return beverage!!.cost() + 0.20 }

}